bots/elasticsearch/tasks.py

```diff
@@ -132,3 +136,73 @@ def serialize(self, source):
             'name': safe_substr(source.long_title),
             'short_name': safe_substr(source.name)
         }
+
+
+class JanitorTask(AppTask):
+    '''
+    Looks for discrepancies between postgres and elastic search numbers
+    Re-indexes time periods that differ in count
+    '''
+    def check_counts_in_range(self, min_date, max_date):
+        partial_database_count = CreativeWork.objects.exclude(title='').exclude(is_deleted=True).filter(
+            date_created__range=[min_date.isoformat(), max_date.isoformat()]
+        ).count()
+        partial_es_count = self.es_client.count(
+            index=(self.es_index or settings.ELASTICSEARCH_INDEX),
+            doc_type='creativeworks',
+            body={
+                'query': {
+                    'range': {
+                        'date_modified': {
+                            'gte': '{}'.format(min_date.isoformat()),
+                            'lte': '{}'.format(max_date.isoformat())
+                        }
+                    }
+                }
+            }
+        )['count']
+        return (partial_database_count == partial_es_count, partial_database_count, partial_es_count)
+
+    def get_date_range_parts(self, min_date, max_date):
+        middle_date = min_date.average(max_date)
+        return {
+            'first_half': {'min_date': min_date, 'max_date': middle_date},
+            'second_half': {'min_date': middle_date, 'max_date': max_date}
+        }
+
+    def pseudo_bisection_method(self, min_date, max_date):
+        '''
+        Checks if the database count matches the ES count
+        If counts differ, split the date range in half and check counts in smaller ranges
+        Pseudo binary because it can't throw away half the results based on the middle value
+        '''
+        from bots.elasticsearch.bot import ElasticSearchBot
+
+        counts_match, db_count, es_count = self.check_counts_in_range(min_date, max_date)
+
+        if counts_match:
+            return
+        if db_count <= 500:
```

Comment:
Should this be `math.abs(db_count - es_count)`?

Hmmm, now that I think about it we may want to make this logic a bit more complex.

I had this backwards when I originally told you this. We should have the cut off be a timedelta to minimize the amount of data we shove back into elastic.

I think it should be more along the lines of:
If the date range you're looking at is less than 12 hours or the total percentage of missing documents is greater than or equal to 60%.